File: gateway/src/config/log_handler.js
import { GATEWAY_GUID } from './config.js';

// =================== LOGGING METHODS ===================
export function logInfo(message) {
    console.log(`\x1b[32m[GATEWAY-${GATEWAY_GUID}] ${message}\x1b[0m`);
}

export function logWarning(message) {
    console.log(`\x1b[33m[GATEWAY-${GATEWAY_GUID}] WARNING: ${message}\x1b[0m`);
}

export function logError(message) {
    console.log(`\x1b[31m[GATEWAY-${GATEWAY_GUID}] ERROR: ${message}\x1b[0m`);
}

export function logMqtt(message) {
    console.log(`\x1b[34m[GATEWAY-${GATEWAY_GUID}] MQTT: ${message}\x1b[0m`);
}

export function logTask(message) {
    console.log(`\x1b[36m[GATEWAY-${GATEWAY_GUID}] TASK: ${message}\x1b[0m`);
}

export function logAuth(message) {
    console.log(`\x1b[35m[GATEWAY-${GATEWAY_GUID}] AUTH: ${message}\x1b[0m`);
}

File: device/lib/log_handler.py

from lib.config import *


def log_revocation(message):
    """Revocation-specific logging"""
    print(f"\033[95m[{DEVICE_GUID}] REVOCATION: {message}\033[0m")


def log_info(message):
    """Clean info logging with device ID"""
    print(f"\033[92m[{DEVICE_GUID}] {message}\033[0m")

def log_warning(message):
    """Clean warning logging with device ID"""
    print(f"\033[93m[{DEVICE_GUID}] WARNING: {message}\033[0m")

def log_error(message):
    """Clean error logging with device ID"""
    print(f"\033[91m[{DEVICE_GUID}] ERROR: {message}\033[0m")

def log_discovery(message):
    """Discovery-specific logging"""
    print(f"\033[94m[{DEVICE_GUID}] DISCOVERY: {message}\033[0m")

def log_key_mgmt(message):
    """Key management logging"""
    print(f"\033[96m[{DEVICE_GUID}] KEY_MGMT: {message}\033[0m")    